.ci/scripts/checkout_complement.sh

```diff
@@ -0,0 +1,20 @@
+#!/bin/bash
+
+mkdir -p complement
+# Attempt to use the version of complement which best matches the current
+# build. Depending on whether this is a PR or release, etc. we need to
+# use different fallbacks.
+#
```

Comment:
Scripts should *start* with a comment saying what they do. Also needs more `set -e`

```suggestion
#!/bin/bash
#
# Fetches a version of complement which best matches the current build.
#
# The tarball is unpacked into `./complement`.

set -e
mkdir -p complement

# Pick an appropriate version of complement. Depending on whether this is a PR or release,
# etc. we need to use different fallbacks:
#
```